test/e2e/cloud_access_role_test.go

```diff
@@ -85,57 +78,73 @@ var _ = Describe("UserLogin", Label("cloud-access-role"), func() {
 	})
 
 	By("Create project with cloud access role", func() {
+		Expect(userData.K8SClient.Get(userData.Context, types.NamespacedName{Name: userData.Project.Name,
+			Namespace: userData.Project.Namespace}, userData.Project)).Should(Succeed())
 		for _, role := range roles {
-			userData.Resources.Project.WithCloudAccessRole(role.AccessRole)
+			userData.Project.Spec.CloudProviderAccessRoles = append(userData.Project.Spec.CloudProviderAccessRoles, role.AccessRole)
 		}
-		actions.PrepareUsersConfigurations(userData)
-		actions.DeployProject(userData)
+		Expect(userData.K8SClient.Update(userData.Context, userData.Project)).Should(Succeed())
 	})
 
 	By("Establish connection between Atlas and cloud roles", func() {
 		Eventually(func() bool {
 			return EnsureAllRolesCreated(*userData, len(roles))
 		}).WithTimeout(5*time.Minute).WithPolling(20*time.Second).Should(BeTrue(), "Cloud access roles are not created")
-		project, err := kube.GetProjectResource(userData)
-		Expect(err).ShouldNot(HaveOccurred())
-		err = cloudaccess.AddAtlasStatementToRole(roles, project.Status.CloudProviderAccessRoles)
+		project := &v1.AtlasProject{}
+		Expect(userData.K8SClient.Get(userData.Context, types.NamespacedName{Name: userData.Project.Name, Namespace: userData.Project.Namespace}, project)).Should(Succeed())
+		err := cloudaccess.AddAtlasStatementToRole(roles, project.Status.CloudProviderAccessRoles)
 		Expect(err).ShouldNot(HaveOccurred())
-		Eventually(kube.GetProjectCloudAccessRolesStatus(userData), "2m", "20s").Should(Equal("True"), "Cloud Access Roles status should be True")
+		Eventually(func() string {
+			return GetProjectCloudAccessRoleStatus(userData)
+		}, "2m", "20s").Should(Equal("True"), "Cloud Access Roles status should be True")
 	})
 
 	By("Check cloud access roles status state", func() {
-		Eventually(kube.GetReadyProjectStatus(userData)).Should(Equal("True"), "Condition status 'Ready' is not 'True'")
-		project, err := kube.GetProjectResource(userData)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(project.Status.CloudProviderAccessRoles).Should(HaveLen(len(roles)))
+		Eventually(func() string {
+			return GetReadyProjectStatus(userData)
+		}).Should(Equal("True"), "Condition status 'Ready' is not 'True'")
+		Expect(userData.K8SClient.Get(userData.Context, types.NamespacedName{Name: userData.Project.Name, Namespace: userData.Project.Namespace}, userData.Project)).Should(Succeed())
+		Expect(userData.Project.Status.CloudProviderAccessRoles).Should(HaveLen(len(roles)))
 	})
 }
 
-func EnsureAllRolesCreated(data model.TestDataProvider, rolesLen int) bool {
-	project, err := kube.GetProjectResource(&data)
-	if err != nil {
-		return false
+func GetReadyProjectStatus(testData *model.TestDataProvider) string {
+	Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{Name: testData.Project.Name, Namespace: testData.Project.Namespace}, testData.Project)).Should(Succeed())
+	for _, condition := range testData.Project.Status.Conditions {
```

Comment:
Such condition ranging "kindly asks" for a general `findCondition` function. Your call thought :)

Reply:
Good point. Done